### PR TITLE
Add remaining metrics for message reliability

### PR DESCRIPTION
**Pull request by adklempner**
Adds the following metrics:
- dial failures
- missed messages
- missed messages relevant to the user
- sent messages which were successfully confirmed

### CODE REVIEW

File: lib/database/sql/000020_dial_failure.up.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    id SERIAL PRIMARY KEY,
+    recordId INTEGER NOT NULL,
+    errorType INTEGER NOT NULL,
+    errorMsg TEXT,
+    protocols TEXT NOT NULL,
+    timestamp INTEGER NOT NULL,
+    CONSTRAINT dialFailure_unique UNIQUE (recordId, errorType, protocols, timestamp)
+);
+
+ALTER TABLE dialFailure ADD CONSTRAINT fk_dialFailure_telemetryRecord
```
Comment (vpavlin) [inline lib/database/sql/000020_dial_failure.up.sql:11]: Shouldn't there be a foreign key constraint for `dial_error_types` as well? Just to make sure we don't insert non-existant error types?

File: lib/database/sql/000020_dial_failure.up.sql [hunk 2]
```diff
@@ -0,0 +1,34 @@
+CREATE TABLE IF NOT EXISTS dialFailure (
+    id SERIAL PRIMARY KEY,
+    recordId INTEGER NOT NULL,
+    errorType INTEGER NOT NULL,
+    errorMsg TEXT,
+    protocols TEXT NOT NULL,
+    timestamp INTEGER NOT NULL,
+    CONSTRAINT dialFailure_unique UNIQUE (recordId, errorType, protocols, timestamp)
```
Comment (vpavlin) [inline lib/database/sql/000020_dial_failure.up.sql:8]: I think this one is missing from the `test_utils.go` cleanup

**Review by vpavlin — APPROVED** [review]
Left some comments, otherwise LGTM

File: lib/database/sql/000020_dial_failure.up.sql (hunk 1 above)
Comment (adklempner) [reply lib/database/sql/000020_dial_failure.up.sql:17]: Added this

File: lib/database/sql/000020_dial_failure.up.sql (hunk 2 above)
Comment (adklempner) [reply lib/database/sql/000020_dial_failure.up.sql]: It seems to work without it but I added it anyways

**Pull request merged**